### PR TITLE
Improvements to the Dark mode and A11y

### PR DESCRIPTION
**Pull request by thetutlage**
This PR contains following changes.

## Dark mode

- Change color contrast for some colors, like sidebar, alert messages, border color and so on.
- Updates the search modal to use dark colors.
- Updates carbon ads to use dark colors.

CleanShot 2022-03-14 at 14 13 21@2x

## A11y improvements

- Use `button` element in place of anchor elements, where the a tag does not change the browser location.
- Use `button` element in place of clickable divs.
- Change tabIndex of heading bookmarks anchor tags. 

## Code cleanup

- Move darkMode frontend code to `app.js` file.
- Cleanup logic for using user preferences over system preferences to be more explicit.

### CODE REVIEW

**Comment by Julien-R44** [comment]
Thanks for taking the time to do these PRs to show me how to do things better. Really appreciate it and takes note of everything.
It is true that I did not produce the cleanest code here, I am not familiar at all with Alpine 😅

LGTM ! Thanks again

**Review by Julien-R44 — APPROVED** [review]

**Comment by thetutlage** [comment]
No worries. That's the goal of these PR's. Finally reach to a stage were our coding styles match :)

File: resources/css/dark-variables.css [hunk 1]
```diff
@@ -5,17 +5,38 @@ html.dark {
...
+  --pre-background: rgb(31 31 31);
   --brand-mute-color: rgb(255 255 255 / 5%);
-  --sidebar-background: var(--background-color);
-  --pre-header-border-color: rgb(64 64 95 / 50%);
+  --sidebar-background: rgb(33 33 33);
+  --pre-header-border-color: rgb(77 77 77 / 50%);
+  --sidebar-shadow: none;
   --pre-header-color: rgb(179 179 208 / 75%);
-  --toc-border-color: #2c2c2c;
+  --toc-border-color: rgb(77 77 77 / 50%);
```
Comment (RomainLanz) [inline resources/css/dark-variables.css:18]:
```suggestion
  --toc-border-color: var(--lighter-gray);
```

File: resources/views/partials/desktop-main-nav.edge [hunk 2]
```diff
@@ -49,7 +49,7 @@
       @endif
     @endeach
     <li class="github-cta-wrapper">
-      <a href="https://github.com/adonisjs/core" target="_blank">
+      <a href="https://github.com/adonisjs/core" target="_blank" title="Github">
```
Comment (RomainLanz) [inline resources/views/partials/desktop-main-nav.edge]: We are also missing some security attributes.

```suggestion
      <a href="https://github.com/adonisjs/core" target="_blank" title="Github" rel="noopener noreferrer nofollow">
```

**Review by RomainLanz — CHANGES_REQUESTED** [review]

**Comment by RomainLanz** [comment]
It seems that we are mixing HEX with RGB color too.

After this PR merged, I would create another one to clean the CSS and migrate everything to HSL. 👍🏻

File: resources/css/dark-variables.css (hunk 1 above)
Comment (thetutlage) [reply resources/css/dark-variables.css:18]: Will leave this for you to fix in another PR

**Pull request merged**